From: Jaromir Dalecky <jaromir@harvestdp.com>
Date: Mon, 07 Nov 2016 23:12:00 +0100
Subject: [PATCH] Request fields from Facebook Graph API

--- a/src/OAuth/UserData/Extractor/Facebook.php
+++ b/src/OAuth/UserData/Extractor/Facebook.php
@@ -24,7 +24,7 @@
     /**
      * Request contants
      */
-    const REQUEST_PROFILE = '/me';
+    const REQUEST_PROFILE = '/me?fields=id,name,email,first_name,last_name,link';
     const REQUEST_IMAGE = '/me/picture?type=large&redirect=false';

     /**
