From: Jaromir Dalecky <jaromir@harvestdp.com>
Date: Tue, 9 Feb 2016 16:57:15 +0100
Subject: [PATCH] Decode JSON response from Facebook API

--- a/src/OAuth/OAuth2/Service/Facebook.php
+++ b/src/OAuth/OAuth2/Service/Facebook.php
@@ -158,20 +158,25 @@
      */
     protected function parseAccessTokenResponse($responseBody)
     {
-        // Facebook gives us a query string ... Oh wait. JSON is too simple, understand ?
-        parse_str($responseBody, $data);
+        $data = @json_decode($responseBody, true);
+        // Facebook gives us a query string on old api (v2.0)
+        if (!$data) {
+            parse_str($responseBody, $data);
+        }

         if (null === $data || !is_array($data)) {
             throw new TokenResponseException('Unable to parse response.');
         } elseif (isset($data['error'])) {
             throw new TokenResponseException('Error in retrieving token: "' . $data['error'] . '"');
+        } elseif (empty($data['access_token'])) {
+            throw new TokenResponseException('Error in retrieving token: Access token is empty!');
         }

         $token = new StdOAuth2Token();
         $token->setAccessToken($data['access_token']);

-        if (isset($data['expires'])) {
-            $token->setLifeTime($data['expires']);
+        if (isset($data['expires_in'])) {
+            $token->setLifeTime($data['expires_in']);
         }

         if (isset($data['refresh_token'])) {
@@ -180,7 +185,7 @@
         }

         unset($data['access_token']);
-        unset($data['expires']);
+        unset($data['expires_in']);

         $token->setExtraParams($data);
